src/tds/codec/token/token_env_change.rs

```diff
@@ -151,82 +154,117 @@ impl TokenEnvChange {
     where
         R: SqlReadBytes + Unpin,
     {
-        let _len = src.read_u16_le().await?;
-        let ty_byte = src.read_u8().await?;
+        let len = src.read_u16_le().await? as usize;
+
+        // We read all the bytes now, due to whatever environment change tokens
+        // we read, they might contain padding zeroes in the end we must
+        // discard.
+        let mut bytes = vec![0; len];
```

Comment:
This is one place where [read-buf](https://github.com/rust-lang/rust/issues/78485) will come in handy — free performance improvement when it's stable.

Reply:
Yeah, except we don't luckily read env tokens that often, and they're all quite short.

Still excited about that RFC.